packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -0,0 +1,137 @@
+import { Provider } from "@ethersproject/abstract-provider";
+
+export enum OptimisticOracleType {
+  PriceRequest = "PriceRequest",
+  Assertion = "Assertion",
+}
+
+export class OptimisticOracleRequest {
+  body: string; // Human readable request body.
+  type: OptimisticOracleType; // Type of the request.
+  timestamp: number; // Timestamp in seconds of the request.
+  identifier: string; // Identifier of the request.
+  requester: string; // Address of the requester.
```

Comment:
Curious to hear your thoughts on an approach to make these _not publicly modifiable_ to enforce immutability.

A few approaches:
1. Apply readonly to all of these.
2. Make store the data field passed to the constructor as a member of the class and make it readonly. Note: the advantage is that it would require less boilerplate to set all these in the constructor and copying to a new struct (when trying to modify) would be much simpler. Downside: accessing the fields (unless you have special getters) would be more annoying `req.data.timestamp` vs req.timestamp`.

We could also just make this softly enforced via code reviews.

Thoughts?

Reply:
Agree, applied readonly